### PR TITLE
Declare out variables inline, and use discards

### PR DESCRIPTION
**Pull request by drieseng**
Declare out variables inline, and use discards when we do not need the value.

### CODE REVIEW

**Comment by drieseng** [comment]
Let me know if you prefer to use **out var**.

**Comment by rynowak** [comment]
> Let me know if you prefer to use out var.

Our guidelines are to use var everywhere that it's possible/feasible.

File: src/Microsoft.AspNetCore.Routing/Constraints/RangeRouteConstraint.cs [hunk 1]
```diff
@@ -58,12 +58,10 @@ public RangeRouteConstraint(long min, long max)
...
             }
 
-            object value;
-            if (values.TryGetValue(routeKey, out value) && value != null)
+            if (values.TryGetValue(routeKey, out object value) && value != null)
             {
-                long longValue;
                 var valueString = Convert.ToString(value, CultureInfo.InvariantCulture);
-                if (Int64.TryParse(valueString, NumberStyles.Integer, CultureInfo.InvariantCulture, out longValue))
+                if (Int64.TryParse(valueString, NumberStyles.Integer, CultureInfo.InvariantCulture, out long longValue))
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Routing/Constraints/RangeRouteConstraint.cs]: nit: Int64.TryParse -> long.TryParse

Comment (drieseng) [reply src/Microsoft.AspNetCore.Routing/Constraints/RangeRouteConstraint.cs]: Done

**Comment by drieseng** [comment]
Should be ok now.
Sorry for the commit/merge mess.

**Pull request merged**